src/overlaybd/tar/erofs/liberofs.cpp

```diff
@@ -580,25 +580,34 @@ static int erofs_init_tar(struct erofs_tarfile *erofstar,
 static int erofs_write_map_file(photon::fs::IFile *fout, uint64_t blksz, FILE *fp)
 {
     uint64_t blkaddr, toff;
-    uint32_t nblocks;
+    uint32_t nblocks, zeroedlen;
+    char *line = NULL;
+    size_t len  = 0;
+    int cnt;
 
     if (fp == NULL) {
        LOG_ERROR("unable to get upper.map, ignored");
        return -1;
     }
     rewind(fp);
-    while (fscanf(fp, "%" PRIx64" %x %" PRIx64 "\n", &blkaddr, &nblocks, &toff)
-           >= 3)
-    {
+
+    while (getline(&line, &len, fp) != -1) {
         LSMT::RemoteMapping lba;
+
+        cnt = sscanf(line, "%" PRIx64" %x %" PRIx64 "%u\n", &blkaddr, &nblocks, &toff, &zeroedlen);
```

Comment:
maybe `"%" PRIx64 "%x%" PRIx64 "%u"` is enough?

Reply:
Done.